packages/compiler-cli/src/ngtsc/typecheck/src/template_symbol_builder.ts

```
@@ -0,0 +1,404 @@
+    } else {
+      return this.getSymbolOfTsNode(node);
+    }
+  }
+
+  private getSymbolOfTsNode(node: ts.Node): ExpressionSymbol|VariableSymbol|ReferenceSymbol|null {
+    if (ts.isParenthesizedExpression(node)) {
+      return this.getSymbolOfTsNode(node.expression);
+    }
+
+    let symbol: ts.Symbol|null;
+    let positionInShimFile: number;
+    if (ts.isPropertyAccessExpression(node)) {
+      symbol = this.typeChecker.getSymbolAtLocation(node.name) ?? null;
+      positionInShimFile = node.name.getStart();
+    } else {
+      symbol = this.typeChecker.getSymbolAtLocation(node) ?? null;
+      positionInShimFile = node.getStart();
+    }
+
+    if (symbol !== null) {
+      const [declarationNode] = symbol.declarations;
+      if (ts.isVariableDeclaration(declarationNode)) {
+        return this.getSymbolOfVariableDeclaration(declarationNode);
+      }
+    }
+
+    const type = this.typeChecker.getTypeAtLocation(node);
+    // If we could not find a symbol, fall back to the symbol on the type for the node.
+    // Some nodes won't have a "symbol at location" but will have a symbol for the type.
+    // One example of this would be literals.
+    symbol = symbol ?? type.symbol ?? null;
+    return {
+      kind: SymbolKind.Expression,
+      tsSymbol: symbol,
+      tsType: type,
+      shimPath: this.shimPath,
+      positionInShimFile
+    };
+  }
+
+  private getSymbolOfVariableDeclaration(declaration: ts.VariableDeclaration): ExpressionSymbol
+      |ReferenceSymbol|VariableSymbol|null {
+    // Instead of returning the Symbol for the temporary variable, we want to get the `ts.Symbol`
+    // for:
+    // - The type reference for `var _t2: MyDir = xyz` (prioritize/trust the declared type)
+    // - The initializer for `var _t2 = _t1.index`.
+    if (declaration.type && ts.isTypeReferenceNode(declaration.type)) {
+      return this.getSymbolOfTsNode(declaration.type.typeName);
+    } else if (declaration.initializer) {
```

else not needed.
I think in this case it's better to reorganize the code to fail fast,
i.e.
```ts
if (!delcaration.initializer) {
  return null;
}
// continue ...
```